### PR TITLE
Milestone 2: Updating and deleting tables

### PR DESCRIPTION
**Pull request by skabeo**
**In this pull request;**

- Added a new column in `schema.sql` file
- Added more data into the `data.sql` file
- Added new queries in the `queries.sql`b file

Screenshot of queries performed are in the comment section below

### CODE REVIEW

**Comment by skabeo** [comment]
#### This comment section shows screenshot of the Queries performed in this pull request

**0. Adding 'unspecified' the species column**

Query 00

**1. Adding species to all animals**

Query 1 PM

**2. Deleting the animals table and rolling back**

Query 1 PM

**3. Performing a SAVEPOINT**



**4. How many animals are there?**



**5. How many animals have never tried to escape ?**



**6. What is the average weight of animals?**



**7. Who escapes the most, neutered or not neutered animals ?**



**8. What is the minimum and maximum weight of each type of animal?**



**9. What is the average number of escape attempts per animal type of those born between 1990 and 2000 ?**

File: queries.sql [hunk 1]
```diff
@@ -33,3 +33,82 @@ WHERE name !== 'Gabumon'
...
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
+-- Deleting the entire table and rolling back
+
+BEGIN;
+
+DELETE FROM animals;
```
Comment (thecodechaser) [inline queries.sql:70]: - [ ] You missed adding the below transaction, Please make sure to add it to the file and a screenshot of the results in the PR.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @skabeo👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️

- [ ] The issues page link in the README file is broken. Please make sure to fix this, It will make your README file more professional.

- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 2]
```diff
@@ -34,6 +34,15 @@ SELECT *
...
 
+-- Adding 'unspecified' to the species
+
+BEGIN;
+
+UPDATE animals;
+SET species = 'unspecified';
+
+ROLLBACK;
+
```
Comment (alexander16108) [inline queries.sql]: - as suggested previously in the requirement you should verify if those changes were made before and after the rollback.  😄

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @skabeo , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights

- Gitflow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Required Changes ♻️
_Check the comments under the review._

- as suggested previously issue page link in the README file is broken. Please make sure to fix this, It will make your README file more professional.
- 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 3]
```diff
@@ -33,3 +33,91 @@ WHERE name !== 'Gabumon'
...
+-- Adding 'unspecified' to the species
+
+BEGIN;
+
+UPDATE animals;
+SET species = 'unspecified';
+
+ROLLBACK;
+
```
Comment (alexander16108) [inline queries.sql]: as i suggested previously in the requirement you should verify if those changes were made before and after the rollback by simply adding 
```
   SELECT species from animals;
   ROLLBACK;

   SELECT species from animals;
```

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @skabeo , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
Gitflow is followed. ✔️
Professional README file. ✔️
PR has a good title and summary. ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @skabeo   👋

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Good job with the queries✔️ 
- Kudus to meet the suggested changes   👍 

## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**